### PR TITLE
Add source-code path adjustment instruction and improve layout of debugging doc [skip ci]

### PR DESCRIPTION
**Pull request by rodnymolina**
Signed-off-by: Rodny Molina <rmolina@nestybox.com>

### CODE REVIEW

File: docs/developers-guide/debug.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ index bb551950..a2b29beb 100644
 
 ```
 
-## Debugger instructions
+## Delve instructions
```
Comment (ctalledo) [inline docs/developers-guide/debug.md:59]: I would leave this as "Debugger" (it's more generic). Recall we use gdb for the C code that is part of sysbox (e.g., nsexec).

File: docs/developers-guide/debug.md [hunk 2]
```diff
@@ -133,87 +132,131 @@ Example:
...
+    (dlv)
+    ```
+
+-   Configure the source-code path:
+
+    If no code is displayed while iterating through the back-trace frames, then
+    ensure that Delve is properly configured with the path to your Sysbox
+    workspace.
+
+    By default, Sysbox expects its source-code to be at `/root/nestybox/sysbox`.
```
Comment (ctalledo) [inline docs/developers-guide/debug.md:190]: This part I don't get; Sysbox does not expect this; it's only that when Sysbox is build inside the build container, it exists as this path.

**Review by ctalledo — APPROVED** [review]
Looks fine, just a couple of comments.

File: docs/developers-guide/debug.md (hunk 1 above)
Comment (rodnymolina) [reply docs/developers-guide/debug.md:59]: I did this to differentiate this section (delve specific) from the other one below (gdb specific). I can reword it as "Delve Debugger Instructions".

File: docs/developers-guide/debug.md (hunk 2 above)
Comment (rodnymolina) [reply docs/developers-guide/debug.md:190]: Yeah, didn't explain myself well there, will rewrite as below ...

```
Sysbox compilation process is carried out inside a docker container. In order to do this, we bind-mount the user's Sysbox workspace (i.e. "sysbox" folder) into this path within the container: `/root/nestybox/sysbox`. Golang compiler includes this path into the generated Sysbox binaries. Thereby, if you are debugging Sysbox daemon in your host system, unless your workspace path fully matches the one above (unlikely), Delve will not be able to display the Sysbox source-code.

The typical solution in these cases is to modify Delve's configuration to replace the containerized path with the one in your local environment.
```

**Pull request merged**